### PR TITLE
Add generate-debug-config.py to tools directory

### PR DESCRIPTION
**Pull request by ben-taussig-solo**
# Description
 - Adds a modified version of upstream's generate_debug_config.py script to the `/tools` directory. This script can be used to generate vscode/gdb debug configuration for an individual test target.
   - This script can be used to work around strange debugger behavior discussed in solo's ["Setting up Ubuntu for Envoy dev" slab guide](https://soloio.slab.com/posts/setting-up-ubuntu-for-envoy-dev-7xfinnwu#thread-236fhosh)
 - More specifically, the script:
   - compiles the test target and a dwarf package file (.dwp) with debug symbols, as described in the envoy gdb guide here: https://github.com/envoyproxy/envoy/blob/main/bazel/README.md#running-a-single-bazel-test-under-gdb
   - updates the user's `.vscode/launch.json` to include a target that can be used to run the test under gdb
 - The original script can be found at: https://github.com/envoyproxy/envoy/blob/main/tools/vscode/generate_debug_config.py
   - Currently, that script is not functional, primarily because it attempts to set launch.json config fields which do not exist, like `debugger_args` (should be miDebuggerArgs)
   - See here: https://github.com/envoyproxy/envoy/blob/main/tools/vscode/generate_debug_config.py#L66-L73
   - The primary changes introduced to the script with this PR correct those fields to point to legitimate launch.json config fields
     - documentation on vscode's launch.json C/C++ config can be found here: https://code.visualstudio.com/docs/cpp/launch-json-reference

### CODE REVIEW

File: tools/generate_debug_config.py [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+        arguments = []
+
+    return {
+        "name": "gdb " + target,
+        "request": "launch",
+        "args": arguments,
+        "type": "cppdbg",
+        "program": str(binary),
+        "miDebuggerArgs": "--directory=" + execroot,
+        "cwd": "${workspaceFolder}",
```
Comment (ben-taussig-solo) [inline tools/generate_debug_config.py:88]: the only changes I made to this file were the block comment at the top of the file, and renaming these fields to refer to valid launch.json fields. Documentation on launch.json config can be found here: https://code.visualstudio.com/docs/cpp/launch-json-reference

File: tools/generate_debug_config.py [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+# Example usage:
+#   `BAZEL_BUILD_OPTION_LIST="--define tcmalloc=gperftools" .tools/generate_debug_config.py //test/extensions/transformers/aws_lambda:api_gateway_transformer_test`
+# This will:
+#   1. Build the target binary with debug info.
+#   2. Generate a launch.json file in the .vscode directory of the workspace, which can be used to debug the target binary with gdb or lldb.
+#   3. The generated launch.json file will have a configuration for the target binary with the specified debugger type.
+#
+# Note: this script expects that the caller is in the root of the workspace
+# Note: use "--overwrite" to overwrite existing config in launch.json. Otherwise, the new config will be appended or added to a backup file in the .vscode directory.
+# Note: I have only tested this script with gdb -- your mileage may vary with lldb.
```
Comment (ben-taussig-solo) [inline tools/generate_debug_config.py:14]: This commentary is the other change I made to the upstream script

File: tools/generate_debug_config.py [hunk 3]
```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/env python3
+
+# Borrowed from upstream: https://github.com/envoyproxy/envoy/blob/main/tools/vscode/generate_debug_config.py
```
Comment (ashishb-solo) [inline tools/generate_debug_config.py:3]: suggestion to change this to a permanent link?

Comment (ben-taussig-solo) [reply tools/generate_debug_config.py]: good call -- updated this here: https://github.com/solo-io/envoy-gloo/pull/318/commits/adb9557f7af9bb1707ca020b43386407b0fbc1df

**Review by ashishb-solo — APPROVED** [review]

**Pull request merged**